test/code/plugins/oms_common_test.rb

```diff
@@ -94,6 +100,256 @@ def teardown
       "OSAlias=UniversalR\n" \
       "OSManufacturer=Central Logistics GmbH\n"
 
+    # Begin:  Unit Tests for some simple methods.
+    # May best be put in another file later.
+
+    def test_clean_hostname_string
+      hn = ' www.five.com:333     '
+      b = Common.clean_hostname_string(hn)
+      expected = 'www.five.com'
+      assert_equal(b,expected,"method clean_hostname_string should return '#{expected}'.")
+      hn = '        microsoft     '
+      b = Common.clean_hostname_string(hn)
+      expected = 'microsoft'
+      assert_equal(b,expected,"method clean_hostname_string should return '#{expected}'.")
+      hn = expected
+      b = Common.clean_hostname_string(hn)
+      assert_equal(b,expected,"method clean_hostname_string should return '#{expected}'.")
+    end
+
+    def test_has_Designated_HostnameFile?
+      file_hostname = '/tmp/containerhostname'
```

Comment:
We should use Ruby's Tempfile utilities to create a file for testing here. On our pBuild machines, if more than one user runs this code, it can collide. That's why I made this change here: https://github.com/Microsoft/OMS-Agent-for-Linux/commit/8bc0525ec5b08c725665d79d69f12feedc199330#diff-3c82e3bf99b11c51027f17b4f05413ddR195